src/type.cpp

```diff
@@ -114,6 +114,8 @@ const Type *Type::ResolveDependenceForTopType(TemplateInstantiation &templInst)
 
 const AtomicType *AtomicType::UniformBool = new AtomicType(AtomicType::TYPE_BOOL, Variability::Uniform, false);
 const AtomicType *AtomicType::VaryingBool = new AtomicType(AtomicType::TYPE_BOOL, Variability::Varying, false);
+const AtomicType *AtomicType::UniformInt1 = new AtomicType(AtomicType::TYPE_INT1, Variability::Uniform, false);
```

Comment:
The implementation of int1 type is not complete in comparison to other types (like int8). I understand that the reason behind it that int1 type is not explicitly visible to users and it has limited usage but it raises lots of questions - why it is added to some methods and not to others, how can we guarantee that it will not break if we add more use cases (intrinsics) that use i1 type?

Reply:
Please add your comment about VaryingInt1 type above to type.h, it'll be very helpful.

Reply:
Done